src/cdk-experimental/ui-patterns/listbox/controller.ts

```diff
@@ -0,0 +1,217 @@
+
+  /** The key used to navigate to the previous item in the list. */
+  prevKey = computed(() => {
+    if (this.state.inputs.orientation() === 'vertical') {
+      return 'ArrowUp';
+    }
+    return this.state.inputs.directionality() === 'rtl' ? 'ArrowRight' : 'ArrowLeft';
+  });
+
+  /** The key used to navigate to the next item in the list. */
+  nextKey = computed(() => {
+    if (this.state.inputs.orientation() === 'vertical') {
+      return 'ArrowDown';
+    }
+    return this.state.inputs.directionality() === 'rtl' ? 'ArrowLeft' : 'ArrowRight';
+  });
+
+  /** The regexp used to decide if a key should trigger typeahead. */
+  typeaheadRegexp = /^.$/; // TODO: Ignore spaces?
+
+  /** The keydown event manager for the listbox. */
+  keydown = computed(() => {
+    const manager = new KeyboardEventManager();
+
+    if (!this.followFocus()) {
+      manager
+        .on(this.prevKey, () => this.prev())
+        .on(this.nextKey, () => this.next())
+        .on('Home', () => this.first())
+        .on('End', () => this.last())
+        .on(this.typeaheadRegexp, e => this.typeahead(e.key));
+    }
+
+    if (this.followFocus()) {
+      manager
+        .on(this.prevKey, () => this.prev({selectOne: true}))
+        .on(this.nextKey, () => this.next({selectOne: true}))
+        .on('Home', () => this.first({selectOne: true}))
+        .on('End', () => this.last({selectOne: true}))
+        .on(this.typeaheadRegexp, e => this.typeahead(e.key, {selectOne: true}));
+    }
+
+    if (this.state.inputs.multiselectable()) {
+      manager
+        .on(Modifier.Shift, ' ', () => this._updateSelection({selectFromAnchor: true}))
+        .on(Modifier.Shift, this.prevKey, () => this.prev({toggle: true}))
+        .on(Modifier.Shift, this.nextKey, () => this.next({toggle: true}))
+        .on(Modifier.Ctrl | Modifier.Shift, 'Home', () => this.first({selectFromActive: true}))
+        .on(Modifier.Ctrl | Modifier.Shift, 'End', () => this.last({selectFromActive: true}))
+        .on(Modifier.Ctrl, 'A', () => this._updateSelection({selectAll: true}));
+    }
+
+    if (!this.followFocus() && this.state.inputs.multiselectable()) {
+      manager.on(' ', () => this._updateSelection({toggle: true}));
+    }
+
+    if (!this.followFocus() && !this.state.inputs.multiselectable()) {
+      manager.on(' ', () => this._updateSelection({toggleOne: true}));
+    }
+
+    if (this.state.inputs.multiselectable() && this.followFocus()) {
+      manager
+        .on(Modifier.Ctrl, this.prevKey, () => this.prev())
+        .on(Modifier.Ctrl, this.nextKey, () => this.next())
+        .on(Modifier.Ctrl, 'Home', () => this.first()) // TODO: Not in spec but prob should be.
+        .on(Modifier.Ctrl, 'End', () => this.last()); // TODO: Not in spec but prob should be.
+    }
+
+    return manager;
+  });
+
+  /** The mousedown event manager for the listbox. */
+  mousedown = computed(() => {
+    const manager = new MouseEventManager();
```

Comment:
Do we need to tear down the old manager when creating a new one?

Reply:
shouldn't need to, the manager doesn't bind itself to any events